roles/job/tasks/main.yml

```diff
@@ -8,7 +16,45 @@
     namespace: "{{ ansible_operator_meta.namespace }}"
   register: k8s_job
 
+- name: Update AnsibleJob isFinished status if job succeeded
+  k8s_status:
+    api_version: tower.ansible.com/v1alpha1
+    kind: AnsibleJob
+    name: "{{ ansible_operator_meta.name }}"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    status:
+      isFinished: true
+  when:
+    - k8s_job['resources'][0]['status']['succeeded'] is defined
+    - k8s_job['resources'][0]['status']['succeeded'] == 1
+
+- name: End play early is AnsibleJob has already finished
+  meta: end_play
+  when:
+    - ansiblejob_info['resources'][0]['status']['isFinished'] is defined
+    - ansiblejob_info['resources'][0]['status']['isFinished']
```

Comment:
Don't we need to refresh the ansiblejob_info value here ? Otherwise we will only have the value from the first read task and the `isFinished: true` update won't be present right ?

Maybe a better scenario would be to put the two tasks (k8s_status and end_play) in a block.

```yaml
- name: update is finished status
  when:
    - k8s_job['resources'][0]['status']['succeeded'] is defined
    - k8s_job['resources'][0]['status']['succeeded'] == 1
  block:
      - k8s_status
      - meta: end_play
```

so we don't have to deal with `ansiblejob_info['resources'][0]['status']['isFinished']`

Reply:
good point, that will make it converge one reconciliation loop sooner.  Thanks!

I am just going to move the task that gets the AnsibleJob info to below the task that update's it's status.  The variable it registers isn't used until after that anyways.  That way we only have one task that ends the play, which might make debugging a little easier.

Reply:
Can confirm that after that change, it ends early now, right after the status is updated.  I checked both scenarios (successful and failed jobs), `isFinished` is still set properly in both scenarios.